jmxtrans-output/jmxtrans-output-kafka/src/main/java/com/googlecode/jmxtrans/model/output/kafka/KafkaWriter.java

```diff
@@ -103,6 +106,7 @@ public KafkaWriter(
 		this.producer= new Producer<String,String>(new ProducerConfig(kafkaProperties));
 		this.topics = asList(Settings.getStringSetting(settings, "topics", "").split(","));
 		this.tags = ImmutableMap.copyOf(firstNonNull(tags, (Map<String, String>) getSettings().get("tags"), ImmutableMap.<String, String>of()));
+		this.typeNamesAsTags = Settings.getBooleanSetting(settings, "typeNamesAsTags", Boolean.FALSE);
```

Comment:
Current code is fairly uncoherent in the way we parse settings. The idea is to get rid of the `settings` Map and rely only on direct attributes of the output writer. Some of the documentation has already been updated, but not yet all of it.